src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,46 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.*;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        BulldozerProducer bulldozerProducer = new BulldozerProducer();
+        TruckProducer truckProducer = new TruckProducer();
+        ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+
+        List<? extends Machine> machines;
+
+        if (type == Bulldozer.class) {
+            machines = bulldozerProducer.get();
+        } else if (type == Truck.class) {
+            machines = truckProducer.get();
+        } else if (type == Excavator.class) {
+            machines = excavatorProducer.get();
+        } else {
+            return List.of();
```

Comment:
As per the checklist, it's better to return an empty list instead of a specific type list when no producer has been found. So, `List.of()` is a good choice here.